crates/core/c8y_api/src/json_c8y.rs

```diff
@@ -117,6 +118,8 @@ impl From<&SoftwareListResponse> for C8yUpdateSoftwareListResponse {
 
         Self {
             c8y_software_list: Some(new_list),
+            // Must derive the source from the topic
+            source: "main".into(),
```

Comment:
This source field **should not** have been implicit like this. It should have been injected/derived from outside. When we want to extend software list support for child devices, this will have to change.

Reply:
Refactored, now it gets the device id from the caller.